locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -1050,42 +1050,59 @@ public void apply(TemplateOptions t, ConfigBag props, Object v) {
-                      for(final Map.Entry<String, String> option : optionsMap.entrySet()) {
-                          Optional<Method> methodOptional = Iterables.tryFind(methods, new Predicate<Method>() {
-                              @Override
-                              public boolean apply(@Nullable Method input) {
-                                  // Matches a method with the expected name, and a single parameter that TypeCoercions
-                                  // can coerce to
-                                  if (input == null) return false;
-                                  if (!input.getName().equals(option.getKey())) return false;
-                                  Type[] parameterTypes = input.getGenericParameterTypes();
-                                  return parameterTypes.length == 1
-                                          && TypeCoercions.tryCoerce(option.getValue(), TypeToken.of(parameterTypes[0])).isPresentAndNonNull();
-                              }
-                          });
-                          if(methodOptional.isPresent()) {
-                              try {
-                                  Method method = methodOptional.get();
-                                  method.invoke(options, TypeCoercions.coerce(option.getValue(), TypeToken.of(method.getGenericParameterTypes()[0])));
-                              } catch (IllegalAccessException e) {
-                                  throw Exceptions.propagate(e);
-                              } catch (InvocationTargetException e) {
-                                  throw Exceptions.propagate(e);
-                              }
-                          } else {
-                              LOG.warn("Ignoring request to set template option {} because this is not supported by {}", new Object[] { option.getKey(), clazz.getCanonicalName() });
-                          }
-                      }
-                  }
+            .put(TEMPLATE_OPTIONS, new CustomizeTemplateOptions() {
+                @Override
+                public void apply(TemplateOptions options, ConfigBag config, Object v) {
+                    if (v == null) return;
+                    @SuppressWarnings("unchecked") Map<String, Object> optionsMap = (Map<String, Object>) v;
+                    if (optionsMap.isEmpty()) return;
+
+                    Class<? extends TemplateOptions> clazz = options.getClass();
+                    Iterable<Method> methods = Arrays.asList(clazz.getMethods());
+                    for(final Map.Entry<String, Object> option : optionsMap.entrySet()) {
+                        Optional<Method> methodOptional = Iterables.tryFind(methods, new Predicate<Method>() {
+                            @Override
+                            public boolean apply(@Nullable Method input) {
+                                // Matches a method with the expected name, and a single parameter that TypeCoercions
```

Comment:
Is the statement "... a single parameter ..." still true, or does this now match methods with multiple parameters?